### PR TITLE
bug 1272643 - start providing workerGroup and workerId value to workers

### PR DESCRIPTION
**Pull request by jhford**
The only open question here is whether to make the workerGroup be the provisionerId or e.g. ec2_us-west-1.  I don't have a strong preference one way or the other, but including the region is 90% of the content of this patch, as it requires an update to the secret entity.

### CODE REVIEW

**Comment by jhford** [comment]
Greg, what do you think about this change?  This will still issue the * scope, but will tell the worker what it *should* be using for workerId and group

File: src/api-v1.js [hunk 1]
```diff
@@ -665,13 +666,17 @@ api.declare({
       token: token,
     });
 
+    let workerId = slugid.v4();
+    let workerGroup = 'ec2_' + secret.region;
```
Comment (jonasfj) [inline src/api-v1.js]: Bike shedding here but with dash it gets hard to parse the workerGroup... So people are less likely to use it in any form of automation... Also it's prettier :)

Regardless this.Secret.load({token}) also works....


Oh, and we need to at-least write the mapping between spotreqid or something and the workerId... Or maybe log requested IP too... Console.log is better than nothing... Queue has a module to get IP from request...

Also what if there is two requests here... There can be due to retires... Do we want to issue the same workerId?
I'm thinking we don't... But maybe we need to spend two braincycles considering the implications...

Comment (jonasfj) [inline src/api-v1.js]: Actually maybe it's better we don't do that in this PR... Probably we should limit number of retires allowed per token...

**Comment by jonasfj** [comment]
What * scopes?

**Comment by jonasfj** [comment]
On topic: why make workgroup = provisionerId? It already given elsewhere...
Having a regions seems nice... As long as it's prefix ec2 :)

File: src/api-v1.js [hunk 2]
```diff
@@ -665,13 +666,17 @@ api.declare({
...
     return res.reply({
       data: secret.secrets,
-      scopes: secret.scopes,
+      workerId: workerId,
+      workerGroup: workerGroup,
       credentials: taskcluster.createTemporaryCredentials({
         scopes: [
           `assume:worker-type:${this.provisionerId}/${secret.workerType}`,
-          'assume:worker-id:*',
+          `assume:worker-id:${workerGroup}:${workerId}`,
```
Comment (jonasfj) [inline src/api-v1.js]: I think this is a legacy scope...

See queue sources it's in the comments of the API definition...
The new line is queue:worker-id:... If I recall correctly..

Comment (jonasfj) [inline src/api-v1.js]: I know it's bad but I can't remove the old one till people stops using it...

File: src/api-v1.js [hunk 3]
```diff
@@ -665,13 +666,17 @@ api.declare({
       token: token,
     });
 
+    let workerId = slugid.v4();
```
Comment (gregarndt) [inline src/api-v1.js]: Is there any length limit on the worker ID from our standpoint since it'll be included in scopes?

File: src/api-v1.js (hunk 1 above)
Comment (gregarndt) [inline src/api-v1.js]: So it looks like we're changing workerGroup to be ec2_<region> instead of just <region> which is ok as far as I'm concerned.  I would like us to maintain region somehow because we use that in metrics and logs for knowing which region a particular set of instances came from.

Comment (jonasfj) [inline src/api-v1.js]: 22 chars... Has to fit in routing key

**Comment by gregarndt** [comment]
This has sat opened for some time now and has merge conflicts too so probably bitrotted a bit.  Do we want to close this for now or should we proceed with trying to implement this?

File: src/api-v1.js (hunk 3 above)
Comment (jonasfj) [inline src/api-v1.js]: Yes, it's in pulse routing keys too..

**Comment by jonasfj** [comment]
Yes, we should be issuing `queue:worker-id:<workerGroup>/<workerId>`

**Comment by djmitche** [comment]
Sounds like we agreed to close this 2 years ago :)
